src/components/task-plan/select-topics.cjsx

```diff
@@ -0,0 +1,136 @@
+      <span className="-section-title">{@props.section.title}</span>
+    </div>
+
+  toggleSection: ->
+    section = @props.section
+    if (TaskPlanStore.hasTopic(@props.planId, section.id))
+      TaskPlanActions.removeTopic(@props.planId, section.id)
+    else
+      TaskPlanActions.addTopic(@props.planId, section.id)
+
+ChapterAccordion = React.createClass
+  displayName: 'ChapterAccordion'
+
+  propTypes:
+    planId: React.PropTypes.any.isRequired
+    courseId: React.PropTypes.any.isRequired
+    chapter: React.PropTypes.object.isRequired
+    hide: React.PropTypes.func.isRequired
+    selected: React.PropTypes.array
+
+  renderSections: (section) ->
+    active = TaskPlanStore.hasTopic(@props.planId, section.id)
+    <SectionTopic active={active} section={section} planId={@props.planId} />
+
+  areAllSectionsSelected: (allSelected, section) ->
+    @props.selected.indexOf(section.id) >= 0 and allSelected
+
+  toggleAllSections: (e) ->
+    if e.target.checked
+      action = TaskPlanActions.addTopic
+    else
+      action = TaskPlanActions.removeTopic
+
+    planId = @props.planId
+    _.each @props.chapter.children, (section) ->
+      action(planId, section.id)
+
+  areAnySectionsSelected: (anySelected, section) ->
+    @props.selected.indexOf(section.id) >= 0 or anySelected
+
+  render: ->
+    chapter = @props.chapter
+    sections = _.map(chapter.children, @renderSections)
+    allChecked = _.reduce(chapter.children, @areAllSectionsSelected, true)
+    expandAccordion = _.reduce(chapter.children, @areAnySectionsSelected, true) or @props.i is 0
+    activeKey = chapter.id if expandAccordion
+
+    header =
+      <h2 className="-chapter-title">
+        <span className="-chapter-checkbox">
+          <input type="checkbox" id="chapter-checkbox-#{chapter.id}"
+            onChange={@toggleAllSections} checked={allChecked}/>
+        </span>
+        <span className="-chapter-number">{chapter.number}</span>
+        <span className="-chapter-title">{chapter.title}</span>
+      </h2>
+
+    <BS.Accordion activeKey={activeKey}>
+      <BS.Panel key={chapter.id} header={header} eventKey={chapter.id}>
+        {sections}
+      </BS.Panel>
+    </BS.Accordion>
+
+SelectTopics = React.createClass
+  displayName: 'SelectTopics'
+  propTypes:
+    planId: React.PropTypes.any.isRequired
+    courseId: React.PropTypes.any.isRequired
+    hide: React.PropTypes.func.isRequired
+    selected: React.PropTypes.array
+
+  getInitialState: ->
+    {courseId} = @props
+    TocActions.load(courseId) unless TocStore.isLoaded()
+    { }
+
+  componentWillMount:   -> TocStore.addChangeListener(@update)
+  componentWillUnmount: -> TocStore.removeChangeListener(@update)
+
+  update: -> @setState({})
+
+
+  renderChapterPanels: (chapter, i) ->
+    <ChapterAccordion {...@props} chapter={chapter}/>
+
+
+  render: ->
+
+    {courseId, planId, selected, hide, header, primary} = @props
+    unless TocStore.isLoaded()
```

Comment:
I was having trouble adding in loadable item here, since TocStore isn't extended from CrudConfig, I wanted to ask you if you think it'd be alright to extend from there?